Prototope/Image.swift

```diff
@@ -34,4 +34,127 @@ public struct Image {
 	init(_ image: UIImage) {
 		uiImage = image
 	}
+    
+    public func toPixels() -> PixelBitmap {
+        return PixelBitmap(image: self)
+    }
 }
+
+public struct PixelBitmap : MutableCollectionType {
+    typealias Index = Int
+    
+    var data: UnsafeMutablePointer<Pixel>
+    let width: Int
+    let height: Int
+    
+    public let startIndex: Int = 0
+    
+    public var endIndex: Int { return height * width }
+    
+    private let dataDestroyer: PixelDataDestroyer
+    
+    init(image: Image) {
+        width = Int(image.size.width)
+        height = Int(image.size.height)
```

Comment:
Careful: `Image.size` reports sizes in points, but for these CG APIs, you'll need to work in pixels.